src/redux/features/location/locationSlice.ts

```diff
@@ -0,0 +1,71 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { CountriesState, Country } from "./types";
+import { ThunkRejectValue } from "../../types";
+import { handleAxiosError } from "../utils/handleThunkError";
+import api from "../../../config/axios";
+
+const initialState: CountriesState = {
+  countries: [],
+  countriesError: null,
+  cities: [],
+  citiesError: null,
+};
+
+export const getCountries = createAsyncThunk<Country[], void, ThunkRejectValue>(
+  "countries/getCountries",
+  async (_, thunkAPI) => {
+    try {
+      const response = await api.get(
+        "https://countriesnow.space/api/v0.1/countries/capital",
+      );
+
+      return [...response.data.data];
```

Comment:
It should call api to your BE and then BE call fake api

Reply:
done